blockchain/blockchain.go

```diff
@@ -599,7 +602,13 @@ func (bc *blockchain) commitBlock(ctx context.Context, blk *block.Block) error {
 	sfTimer := bc.timerFactory.NewTimer("sf.Commit")
 	err = bc.sf.Commit(ctx, blk)
 	sfTimer.End()
-	// detach working set so it can be freed by GC
+	if errors.Cause(err) == factory.ErrNotValidated {
```

Comment:
this is kine of strange, Commit() --> ErrNotValidated --> then re-commit?
can there be a better way?

Reply:
can we nil the workingset, when validation fail?

factory can regenerate the workingset, but it will fail upon Commit()